main.go

```diff
@@ -1,12 +1,149 @@
 package main
 
+import (
+	"fmt"
+	"os"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/microkit/command"
+	microserver "github.com/giantswarm/microkit/server"
+	"github.com/giantswarm/microkit/transaction"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/microstorage"
+	"github.com/giantswarm/microstorage/memory"
+	"github.com/spf13/viper"
+
+	"github.com/giantswarm/cluster-operator/flag"
+	"github.com/giantswarm/cluster-operator/server"
+	"github.com/giantswarm/cluster-operator/service"
+)
+
+const (
+	notAvailable = "n/a"
+)
+
 var (
-	description string = "The cluster-operator is an in-cluster agent that handles Giant Swarm guest cluster specific resources."
-	gitCommit   string = "n/a"
-	name        string = "cluster-operator"
-	source      string = "https://github.com/giantswarm/cluster-operator"
+	description            = "The cluster-operator is an in-cluster agent that handles Giant Swarm guest cluster specific resources."
+	f           *flag.Flag = flag.New()
+	gitCommit              = notAvailable
+	name                   = "cluster-operator"
+	source                 = "https://github.com/giantswarm/cluster-operator"
 )
 
 func main() {
+	err := mainWithError()
+	if err != nil {
+		panic(fmt.Sprintf("%#v\n", err))
+	}
+}
+
+func mainWithError() (err error) {
+
+	// Create a new logger that is used by all packages
+	var newLogger micrologger.Logger
+	{
+		loggerConfig := micrologger.DefaultConfig()
+		loggerConfig.IOWriter = os.Stdout
+		newLogger, err = micrologger.New(loggerConfig)
+		if err != nil {
+			return microerror.Maskf(err, "micrologger.New")
+		}
+	}
+
+	// Define server factory to create the custom server once all command line
+	// flags are parsed and all microservice configuration is processed
+	newServerFactory := func(v *viper.Viper) microserver.Server {
+		// New custom service implements the business logic
+		var newService *service.Service
+		{
+			serviceConfig := service.DefaultConfig()
+
+			serviceConfig.Flag = f
+			serviceConfig.Logger = newLogger
+			serviceConfig.Viper = v
+
+			serviceConfig.Description = description
+			serviceConfig.GitCommit = gitCommit
+			serviceConfig.Name = name
+			serviceConfig.Source = source
+
+			newService, err = service.New(serviceConfig)
+			if err != nil {
+				panic(fmt.Sprintf("%#v\n", microerror.Maskf(err, "service.New")))
+			}
+
+			// TODO: Comment the lifecycle of this goroutine
```

Comment:
Here is a TODO left. I think such docs would be good to have at the `Boot` method. Then it is visible in the godoc and actually useful for users.